standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java

```diff
@@ -3380,8 +4765,68 @@ public GetAllFunctionsResponse getAllFunctions() throws TException {
     return client.get_all_functions();
   }
 
+  protected void create_table(CreateTableRequest request) throws TException {
+    client.create_table_req(request);
+  }
+
+  protected void drop_table_with_environment_context(String catName, String dbname, String name,
+      boolean deleteData, EnvironmentContext envContext) throws TException {
+    DropTableRequest dropTableReq = new DropTableRequest(dbname, name);
+    dropTableReq.setDeleteData(deleteData);
+    dropTableReq.setCatalogName(catName);
+    dropTableReq.setDropPartitions(true);
+    dropTableReq.setEnvContext(envContext);
+    dropTableReq.setAsyncDrop(!isLocalMetaStore());
+    TableOpResp resp = client.drop_table_req(dropTableReq);
+    dropTableReq.setId(resp.getId());
+    try {
+      while (!resp.isFinished() && !Thread.currentThread().isInterrupted()) {
```

Comment:
Should we add a timeout? Otherwise, we would repeatedly hammer HMS about dropping the table.

Reply:
The client can interrupt the current thread to terminate the drop